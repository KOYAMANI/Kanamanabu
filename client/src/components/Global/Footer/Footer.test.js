import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Footer from "./Footer";

afterEach(cleanup);

describe('Footer', () =>{

    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<Footer/>, div)
    });

    test('Render text', ()=>{
        render(<Footer/>)
        const element = screen.getByText('Copyright © Kanamanabu')
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent('Copyright © Kanamanabu')
    });

    test('match snapshot',()=>{
        const tree = renderer.create(<Footer/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})