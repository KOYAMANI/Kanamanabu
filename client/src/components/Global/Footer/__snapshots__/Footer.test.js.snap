// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer match snapshot 1`] = `
<footer
  style={
    Object {
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-center py-3 col"
      >
        <p
          className="text-center text-black"
        >
          Copyright Â© Kanamanabu
        </p>
      </div>
    </div>
  </div>
</footer>
`;
